### ISSUE TITLE
200K CVEs - extract Keyphrases

### ISSUE THREAD

**Issue by Crashedmind** [CyberSecAI/CWEMap#48]
1. **User Scenario**: https://github.com/CyberSecAI#extract-key-phrases-from-vulnerability-description
2. **Repo**: https://github.com/CyberSecAI/cve_info
3. **Demo**: https://github.com/CyberSecAI/cve_info see examples in readme

**Related Tickets**
1. https://github.com/CyberSecAI/CWEMap/issues/48
2. https://github.com/CyberSecAI/CWEMap/issues/13
3. https://github.com/CyberSecAI/CWEMap/issues/52

- [x] Extract the KeyPhrases for 25OK CVEs
- [ ] Use the data to reinforce dataset

separate ticket
- [ ] train model with new dataset

https://github.com/CyberSecAI/cve_info now contains the keyphrases extracted for ~55K CVEs across these populations:

- CISA KEV ~1.5K
- MITRE Top 25 ~7K
- CISA Vulnrichment ~46.5 K

I did not yet add the mitreTechnicalImpacts field to the CISA Vulnrichment population (because I was not satisfied with the results at this scale).

- Specifically, for some CVE Descriptions and associated extracted impact keyphrase there isn't a strong link to a specific mitreTechnicalImpacts.
- I made mitreTechnicalImpacts optional in the schema

I decided I will proceed as follows:

1. Extract the KeyPhrases from the remaining ~200K CVEs
         a. Without additional model finetuning i.e. model is performing well enough (based on LLM + manual review)
3. Validate and Augment the ~250K KeyPhrases with different approaches
    a. https://huggingface.co/datasets/yahoo-inc/cve-impacts which used GliNER for all published CVEs 
    b. Apply a traditional (regex + fuzzy) approach with common known good phrases in parallel to validate and augment results on ~250K CVEs.
              i. My experience to data in general is that it's best to combine the best of both worlds.
4. Assign mitreTechnicalImpacts
5. Document all this in https://cybersecai.github.io/
6. Productionize this e.g. make it a github action that daily processes new (or unprocessed) CVEs from https://github.com/CVEProject/cvelistV5

**Comment by Crashedmind** [comment]
1. review CVE-2003-1010_keyphrases.json  "rootcause": "Unknown vulnerability",

**Comment by Crashedmind** [comment]
Remove and redo the descriptions without  **keyphrases**
 DataFrame with 180321 rows

DataFrame Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180321 entries, 0 to 180320
Data columns (total 7 columns):
 #   Column                 Non-Null Count   Dtype 
---  ------                 --------------   ----- 
 0   cveId                  180321 non-null  object
 1   version                180321 non-null  object
 2   timestamp              180321 non-null  object
 3   description            180321 non-null  object
 4   **keyphrases**             179571 non-null  object
 5   source_file            180321 non-null  object
 6   mitreTechnicalImpacts  7203 non-null    object
dtypes: object(7)
memory usage: 9.6+ MB
None

**Comment by Crashedmind** [comment]
Publish the KeyPhrase Extraction Code

**Comment by Crashedmind** [comment]
fix https://nvd.nist.gov/vuln/detail/CVE-2022-37767

**Comment by Crashedmind** [comment]
these have tags in https://github.com/CVEProject/cve-schema/blob/main/schema/CVE_Record_Format.json:
[product](https://github.com/CVEProject/cve-schema/blob/a29f28e5d48383cc5e179f9c6655ac49e8ffe1f9/schema/CVE_Record_Format.json#L106),
[version](https://github.com/CVEProject/cve-schema/blob/a29f28e5d48383cc5e179f9c6655ac49e8ffe1f9/schema/CVE_Record_Format.json#L284),
[components](https://github.com/CVEProject/cve-schema/blob/a29f28e5d48383cc5e179f9c6655ac49e8ffe1f9/schema/CVE_Record_Format.json#L221)
[impact](https://github.com/CVEProject/cve-schema/blob/a29f28e5d48383cc5e179f9c6655ac49e8ffe1f9/schema/CVE_Record_Format.json#L621)
These don't
vector
rootcause, weakness
[cweId](https://github.com/CVEProject/cve-schema/blob/a29f28e5d48383cc5e179f9c6655ac49e8ffe1f9/schema/CVE_Record_Format.json#L886) would be what these would map to 
There is some discussion about root cause tags https://github.com/CVEProject/cve-schema/pull/335/commits/6b3c524fdde009d7ff8bba9ce4c41ad24f0cf336

**Comment by Crashedmind** [comment]
- [ ] CVE-1999-0007,1.0.0,2024-12-10T17:54:04.992951+00:00,Information from SSL-encrypted sessions via PKCS #1.,0,,,,,,,,,
